src/Http/Uri.php

```diff
@@ -330,7 +330,7 @@ public function idn()
      * or any other executed script.
      *
      * @param array $props
-     * @return string
+     * @return string|static
```

Comment:
It never actually returns string though, does it?

```suggestion
     * @return static
```